content/docs/command-reference/plots/index.md

```diff
@@ -107,7 +98,144 @@ names in the `train` array below:
 }
 ```
 
-## Plot templates (data series only)
+## Configuring a plot
+
+In order to create visualizations, users need to provide the data and
+(optionally) configuration that will help customize the plot. DVC provides two
+ways to configure visualizations. Users can mark specific <abbr>stage</abbr>
+<abbr>outputs</abbr> as plot or define plot configuration inside `dvc.yaml`
+under `plots` key.
+
+### Top-level plot definitions
+
+Plots can be defined in `dvc.yaml` under the `plots` key. Unlike
+[stage plots](#stage-plots), these are especially useful when users want to
+compare data from different data sources residing on the same version of the
+project. For example, comparing training versus test results on current branch.
```

Comment:
> They enable covering **multiple data sources in one plot** or **using the same data source in multiple plots**...
> The main point is that they can **use data sourced from different files**, maybe we should include that?

Also found more specific context in https://github.com/iterative/dvc/issues/7086#issuecomment-985640176:
> 1. iterative/dvc#6316 (single-source segmentation)
> 2. Multiple series per axis (single-source comparison)
> 3. Multiple files within one plot (multi-source composite)

... Any more?

We should expand a bit on the benefits/features of top-level definitions (which may soonish become the only documented place to define plots per https://github.com/iterative/dvc.org/issues/3763). "to compare data from different data sources residing on the same version of the project" doesn't seem to cover all of it.

Reply:
I think that this one sentence pretty much covers all of it, but I agree that we might not be doing the best job explaining its value here. Let me think how to modify that.

Reply:
@jorgeorpinel As you mention above, we shouldn't be comparing to "stage plots" if the "top-level" ones are the default and soon the only supported method. Do we need to explain their unique benefits in that case? Maybe we instead need to explain "stage plots" as a subset that can be defined inside a pipeline stage but are limited to configuring a single output as a plot?

More importantly, should we continue to discuss in this PR or is the rest of it in good enough shape that we could merge and iterate on top?

Reply:
> > compare data from different data sources
>
> this one sentence pretty much covers all of it

@pared what about the same data source file for 2+ plots?

> Do we need to explain their unique benefits?

@dberenbaum right, I didn't mean the benefits over stage-level plots but rather just the benefits/features of plot definitions (which are top-level).

> explain "stage plots" as a subset

Best not to worry about updating those existing explanations until #3767 is addressed. This PR is already complex.

> should we continue to discuss in this PR or is the rest of it in good enough shape

This question shouldn't a blocker but we should probably make a summary of pending questions and try to get this approved... ⌛

Reply:
> @pared what about the same data source file for 2+ plots?

You got me here, yes same data source file can also be used